benchmark.R

```diff
@@ -0,0 +1,182 @@
+europe <- countries %>% 
+  filter(continent == "Europe") %>% 
+  st_union()
+
+#' ## Check sPlotOpen documentation
+
+#' Check for package information
+?sPlotOpenR
+
+#' There is no general description of the package.
+
+??sPlotOpenR
+#' But there is good documentation for the functions.
+
+
+#' ## Testing the functions
+#' ### get_sPlot()
+#'
+#' With default options
+get_sPlot()
+
+#' It worked nicely, including the progress messages and the creation of the directories.
+#' However, all files in the iDiv repository were saved within the directory, in addition to the three table data (DT, header and CWM_CVM)
+#' For example, the pdf file Demo.pdf and the folder Schema. 
+#' I believe this is occurring because now we can only download all files from the iDiv portal.
+#' Also, I think this will be changed when the  argument metadata is implemented, right?
+#'
+#'
+#' #### Changing arguments
+#' 
+#' 
+#' Ask for nonexistent table
+get_sPlot(tables = "test")
+#' Correct error message
+#' 
+#' Ask for nonexistent table while also asking for existing table
+get_sPlot(tables = c("header", "test"))
+#' Correct message about overwriting header, but no message that "test" was ignored. Should we add it?
+#'
+#'
+#' Include existent directory
+get_sPlot(dir = "C:/Users/faria/Nextcloud/sPlot")
+#' It works nice, but save all files directly in directory, without placing them in a folder.
+#' I think an option to handle this is to create a folder named after the version of sPlotOpen being downloaded, as discussed in the Google Docs.
+#'
+#'
+#' Include nonexistent directory
+get_sPlot(dir = "C:/Users/faria/Nextcloud/nonexisting")
+#' Works nicely, creating the desired directory.
+#' 
+#' 
+#' Only download files, without loading them in R.
+get_sPlot(load = FALSE)
+#' Works nicely.
+#' 
+#' Not requesting metadata
+get_sPlot(metadata = FALSE)
+#' Not implemented yet.
+#' 
+#' 
+#' ### read_sPlot()
+#' 
+#' With default options
+#' 
+read_sPlot()
+#' Works fine.
+#' 
+#' Reading from an alternative directory
+read_sPlot(dir = "C:/Users/faria/Nextcloud/sPlotOpen/data")
+#' Works fine.
+#' 
+#' Reading it from a wrong directory.
+read_sPlot(dir = "~/Nextcloud")
+#' It simply does nothing, should we add an error message?
+#'
+#' Reading only one table
+read_sPlot(tables = c("DT"))
+#' Works fine.
+#' 
+#' Reading a nonexistent table
+read_sPlot(tables = c("test"))
+#' Correct error message
+#' 
+#' Reading a nonexistent table together with an existent table
+read_sPlot(tables = c("DT", "test"))
+#' No message about the ignored argument. Should we add any?
+#' 
+#' 
+#' 
+#' ### filter_species()
+#' 
+#' Default options
+spp_list <- c("Luzula arcuata", "Bauhinia variegata")
+data <- read_sPlot()
+
+filter_species(data, spp_list)
+#' Works fine, including the messages.
+#' 
+#' Setting resolve = TRUE
+filter_species(data, spp_list = c("Luzulula arcuata"), resolve = TRUE)
+#' **Dependency problem: results in error if TNRS package is not installed.**
```

Comment:
I wanted to avoid having lots of dependencies, since that makes it more difficult to install the package. I included TNRS and dggridR as "Suggests" rather than "Imports" for this reason, but maybe it annoying for the user to get an error message when they try to use the function. Let me know your thoughts.

Reply:
I also think it is better to keep the package without dependencies. One suggestion would be to add an error message like: "You need to have the package TNRS installed for performing the taxonomic resolution. Please install it before you set the argument resolve = TRUE."

One way of doing it would be:

if(!require(TNRS), stop(<message>))